skorch/tests/callbacks/test_training.py

```diff
@@ -317,3 +317,276 @@ def test_wrong_threshold_mode(
 
         expected_msg = "Invalid threshold mode: 'incorrect'"
         assert exc.value.args[0] == expected_msg
+
+
+
+class TestParamMapper:
+
+    @pytest.fixture
+    def initializer(self):
+        from skorch.callbacks import Initializer
+        return Initializer
+
+    @pytest.fixture
+    def freezer(self):
+        from skorch.callbacks import Freezer
+        return Freezer
+
+    @pytest.fixture
+    def unfreezer(self):
+        from skorch.callbacks import Unfreezer
+        return Unfreezer
+
+    @pytest.fixture
+    def param_mapper(self):
+        from skorch.callbacks import ParamMapper
+        return ParamMapper
+
+    @pytest.fixture
+    def net_cls(self):
+        from skorch import NeuralNetClassifier
+        return NeuralNetClassifier
+
+    @pytest.mark.parametrize('weight_pattern', [
+        'dense*.weight',
+        lambda name: name.startswith('dense') and name.endswith('.weight'),
+    ])
+    def test_initialization_is_effective(self, net_cls, classifier_module,
+                                         classifier_data, initializer,
+                                         weight_pattern):
+        from torch.nn.init import constant_
+        from skorch.utils import to_numpy
+
+        net = net_cls(
+            classifier_module,
+            lr=0,
+            max_epochs=1,
+            callbacks=[
+                initializer(weight_pattern, partial(constant_, val=5)),
+                initializer('dense1.bias', partial(constant_, val=10)),
+            ])
+
+        net.fit(*classifier_data)
+
+        assert np.allclose(to_numpy(net.module_.dense0.weight), 5)
+        assert np.allclose(to_numpy(net.module_.dense1.weight), 5)
+        assert np.allclose(to_numpy(net.module_.dense1.bias), 10)
+
+    def test_initialization_is_effective_pre(self, net_cls, classifier_module,
+                                             classifier_data, initializer):
+        from torch.nn.init import constant_
+        from skorch.utils import to_numpy
+
+        mod = classifier_module()
+
+        net = net_cls(
+            mod,
+            lr=0,
+            max_epochs=1,
+            callbacks=[
+                initializer('dense*.weight', partial(constant_, val=5)),
```

Comment:
I believe this pattern will fail now with the refactored tests because `classifier_module` now uses `nn.Sequential` internally.